src/main/java/br/com/zup/library/emprestimo/EmprestimoService.java

```
@@ -0,0 +1,69 @@
+package br.com.zup.library.emprestimo;
+
+import br.com.zup.library.exemplar.Exemplar;
+import br.com.zup.library.exemplar.ExemplarRepository;
+import br.com.zup.library.livro.Livro;
+import br.com.zup.library.usuario.Usuario;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
+
+import static br.com.zup.library.exemplar.TipoCirculacao.LIVRE;
+import static br.com.zup.library.usuario.TipoUsuario.PADRAO;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@Service
+public class EmprestimoService {
+
+    private final EmprestimoRepository emprestimoRepository;
+    private final ExemplarRepository exemplarRepository;
+
+    @Autowired
+    public EmprestimoService(EmprestimoRepository emprestimoRepository,
+                             ExemplarRepository exemplarRepository) {
+        this.emprestimoRepository = emprestimoRepository;
+        this.exemplarRepository = exemplarRepository;
+    }
+
+    @Transactional
+    public NovoEmprestimoResponse realizaEmprestimo(SolicitacaoEmprestimo request) {
+        var usuario = request.getUsuario();
+        var livro = request.getLivro();
+
+        var exemplar = buscaExemplar(livro, usuario);
+
+        validaNumeroMaximoEmprestimos(usuario);
```

aqui vc continua carregando o exemplar à toa. a validação vem depois quando na verdade seria mais apropriado vir antes.

lembre-se do **fail fast**.

---

resolvido